blocks/pipeline.py

```diff
@@ -245,13 +245,13 @@ def list_to_dict(obj: List):
         if hasattr(el, "id")
         else type(el).__name__: obj_to_dict(el)
         if is_custom_obj(el)
-        else el
+        else copy(el)
         for el in obj
     }
 
 
-def obj_to_dict(obj: Any) -> dict:
-    obj_dict = vars(copy(obj))
+def obj_to_dict(obj: Any):
+    obj_dict = copy(vars(obj))
```

Comment:
hmm, why did this change? and why is the return type removed?
I actually don't know whether it makes sense to copy the dictionary. have you run this through an sklearn pipeline and verified it still works?

Reply:
Why doesn't it make sense to copy the dictionary? 
What we are doing here is: converting the object to a dictionary, then looping through the value of that dictionary and expanding it's child objects if they are expandable (not built in types).

I'm adding the type annotation back.

Reply:
until we have integration tests, I can't verify that this is working. if you want to change it, please open a PR when we have integration tests, so we can automatically check whether this breaks something. sorry, I spent last night fixing bugs. don't want to do that today as well.

Reply:
I can safely say the previous version was working, since that's what I'm running on the VM.

Reply:
🤷 Ok, but at least add the copying of the list below.

Reply:
according to the title of the PR, this should only contain that change, yes!